tests/integration/save_test.js

```diff
@@ -465,6 +465,39 @@ module('integration - Persistence', function(hooks) {
     });
   });
 
+  test('handles nullability from server', async function(assert) {
+    let data = {
+      name: {
+        first: 'Eddard',
+        last: 'Stark'
+      }
+    };
+
+    let person = store.push({
+      data: {
+        type: 'person',
+        id: 1,
+        attributes: data
+      }
+    });
+
+    return run(() => {
+      assert.equal(person.name.first, 'Eddard');
+
+      person = store.push({
+        data: {
+          type: 'person',
+          id: 1,
+          attributes: {
+            name: null
+          }
+        }
+      });
+
+      assert.equal(person.name, null);
```

Comment:
@ro0gr This looks like a good change. Any chance you could update the test to test `person.names` too so that we know this works for fragment arrays too?

Reply:
sure, will do!

Reply:
@patocallaghan thanks for the suggestion. I've improved tests a bit.